src/StateMachine.jl

```diff
@@ -305,11 +293,7 @@ function deduplicate_state_machine(entry::CodePoint, state::BinderState)
     top_down_states = reachable_states(entry)
     bottom_up_states = Iterators.reverse(top_down_states)
     for e in bottom_up_states
-        d = dedup(dedup_map, e, state)
-        if d.label === nothing
-            # It is a newly seen state
-            push!(result, d)
-        end
+        dedup(dedup_map, e, state)
```

Comment:
should `dedup` be `dedup!` - it looks like we are relying on mutation here as the main effect. It also looks like `result` isn't used anymore